system/datatypes/mailinglist.php

```diff
@@ -1083,6 +1151,25 @@ protected function mailingSendTestAction(HyphaRequest $request) {
 		return 'reload';
 	}
 
+	protected function mailingProgressAction(HyphaRequest $request) {
+		if (!$this->hasSender()) {
+			notify('error', __('ml-no-sender'));
+			return null;
+		}
+
+		$mailingId = $request->getArg(1);
+
+		// check if given id was correct
+		/** @var HyphaDomElement $mailing */
+		$mailing = $this->xml->getElementById($mailingId);
+		if (!$mailing instanceof HyphaDomElement) {
+			return '404';
+		}
+		$progress = $mailing->getAttribute('progress').'/'.$this->getNumberOfAddresses();
```

Comment:
Would it be better to return JSON instead of a 0/0 format that needs to be split client-side? JSON is easier and more robust to parse?

Reply:
We can do that. Better/easier is a bit subjective  :-) 

JSON is used because is it both human readable and computer parsable. If find `'5/10'` more readable than `{"current":5,"total":10}`. 
Parsing is `const {current,total} = JSON.parse(string);` vs `const [current,total] = string.split('/');` not much difference in complexity there.

And as to robustness `(x,y) => x+'\'+y` is a fully bijective function as for x,y without slashes so parsing is as robust as it gets (and probably even faster than JSON parse)

But I'm more than fine to switch to JSON :-)

Reply:
Agreed, but JSON is also more extendable (and if we add other AJAX APIs, can be used more consistently), so I'd still favor JSON here.

Reply:
Agreed :)

Reply:
And done